<?xml version="1.0"?>
<launch>
	<!-- Input arguments -->
	<arg name="argID"               default="0"/>
	<arg name="argSafety"           default="false"/>
	<arg name="argTakeOffAltitude"  default="10.0"/>
	<arg name="argTakeOffThreshold" default="0.3"/>
	<arg name="argTargetFrame"      default="base_link"/>
	<arg name="argSourceFrame"      default="odom"/>
	<arg name="argNodeRate"         default="100.0"/>
	<arg name="argNamespace"        default="heifu$(arg argID)"/>
	<arg name="argVehicle"          default="$(arg argNamespace)"/>
	<arg name="argGoalTolerance"    default="1.0"/>
	<arg name="argRRTStep"          default="0.25"/>
	<arg name="argBiasedRRT"        default="false"/>
	<arg name="argBiasedRRTProb"    default="0.50"/>

	<!-- Launch mavros node -->
	<node name="rrt_node" type="rrt_node" pkg="rrt" output="screen" required="true" ns="$(arg argNamespace)">
		<param name="paramTargetFrame"      type="string" value="$(arg argNamespace)/$(arg argTargetFrame)"	/>
		<param name="paramMapFrameId"       type="string" value="$(arg argNamespace)/$(arg argSourceFrame)"	/>

		<param name="paramNodeRate"         type="double" value="$(arg argNodeRate)"		/>
		<param name="paramGoalTolerance"	type="double" value="$(arg argGoalTolerance)"	/>
		<param name="paramRRTStep"			type="double" value="$(arg argRRTStep)"			/>
		<param name="paramBiasedRRTProb"	type="double" value="$(arg argBiasedRRTProb)"	/>

		<param name="paramBiasedRRT"		type="bool" value="$(arg argBiasedRRT)"	/>

		<rosparam param="paramRobotSize"		>[ 1.0,  1.0,  1.0]</rosparam>
		<rosparam param="paramWorkspaceSize"	>[20.0, 20.0, 10.0]</rosparam>
		<rosparam param="paramWorkspaceCenter"	>[ 0.0,  0.0,  5.0]</rosparam>
		<rosparam param="paramGoalPosition"		>[ 0.0, 32.0,  14.0, 0.0, 0.0, 0.0]</rosparam>
	</node>
</launch>

